### DISCUSSION TITLE
Hello old friend, Binding to Realm IList issue #1545

### DISCUSSION THREAD

**Discussion by qwertylolman** [MvvmCross/MvvmCross#4089] [Q&A]
I'm not sure if this is a MvvmCross issue. so I don't want to create a new bug. but it's 2021, bug was closed in 2017, and issue still can be reproduced.
original issue https://github.com/MvvmCross/MvvmCross/issues/1545

so from time to time this issue appears in my projects, as i'm using Realm and MvvmCross. I'm not using direct bindings to realm objects, everything is wrapped into viewmodels. is someone found another fix except of using the MvxBindingContext? i've tried:
- detached copy of realm object
- Freeze function

both of those cases should be thread safe (according to realm documentation). still, no luck. 
any ideas? recently, this issue was found in case below:
`FullName => entity = Entity; entity.FirstName + entity.LastName`
output was only the FirstName

--
my typical layers looks like this:

    AwesomeViewModel
    {
        Init() => do some fetch and wrap data into ItemViewModel, put everyting into ObservableCollection
    }

    AwesomeItemViewModel
    {
        Entity => fetch entity from realm. configurable. can fetch new entity from realm for every call, or can create a snapshot of entity on first call, detach from realm (freeze or create a copy of an object) and return cached value
        Name => Entity.Name
    }

    Fluent Bindings on platforms
    {
        Bind(TextView).To(BindText).For(Name)
    }

**Comment by Cheesebaron** [comment]
Please provide more context. I don't remember every issue about every edge case.

What is the issue to begin with? NotifyPropertyChanged doesn't fire on Realm objects?

**Reply by qwertylolman** [reply]
https://github.com/MvvmCross/MvvmCross/issues/1545
looks like events are fired, but due to some threading we have missed data.
MvvmCross 7.1.2 Realm 5.1.2

**Reply by Cheesebaron** [reply]
I don't understand what "missed data" means. Does it mean that:
1. UI doesn't update
2. Models somehow deadlock and don't have their data populated
3. Something else?
